tudothesis.cls

```diff
@@ -61,6 +61,29 @@
 \let\oldtitle\title%
 \renewcommand{\title}[1]{\oldtitle{#1}\newcommand{\thetitle}{#1}}
 
+% Provide all translations through the "translations" package
+\RequirePackage{translations}
+
+\DeclareTranslation{German}{thesisclass}{Arbeit zur Erlangung des akademischen Grades}
+\DeclareTranslation{English}{thesisclass}{Thesis for obtaining the academic degree}
```

Comment:
These are not the usual translations of these terms, I don't know if there is an official source for english terms though.